[Decomposition] full_like

salilsdesai opened the pull request:
Summary: Moving decomposition from _inductor/decomposition.py and include it in core_aten_decompositions

Test Plan: Phabricator + OSS Tests

Differential Revision: D48939842




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @Xia-Weiwen @ngimel

peterbell10 commented on torch/_decomp/decompositions.py:
```diff
@@ -3980,6 +3980,28 @@ def scaled_dot_product_flash_attention(
+    memory_format=torch.preserve_format,
+):
+    return torch.full(
+        [*self.size()],
```
This is wrong because `torch.full_like` should preserve the stride order of `self` in the same way that `empty_like` does. This decomposition works fine for inductor though because we don't care about strides.

salilsdesai replied:
@peterbell10 So what can I do to preserve stride order? I tried using (...).as_strided(...) but it did not help

salilsdesai committed:
[Decomposition] full_like (#108537)

Summary:
bypass-github-pytorch-ci-checks


Moving decomposition from _inductor/decomposition.py (plus modifying it to use memory_format) and include it in core_aten_decompositions

Test Plan: Phabricator + OSS Tests

Differential Revision: D48939842
